staging/src/k8s.io/kubectl/pkg/cmd/set/set_env.go

```diff
@@ -305,6 +305,12 @@ func (o *EnvOptions) RunEnv() error {
 			case *v1.Secret:
 				for key := range from.Data {
 					if contains(key, o.Keys) {
+						for _, l := range key {
```

Comment:
You will probably want to pull this into it's own function.

Reply:
Agreed.  What about just using `keyToEnvName()`, and print a warning anytime keyToEnvName(key) != key?  I don’t think there is a need to iterate over each character.

Reply:
a warning added into `keyToEnvName()`

Reply:
unit cases have been added acoordingly